src/liballoc/vec.rs

```diff
@@ -2751,6 +2752,7 @@ pub struct DrainFilter<'a, T, F>
     del: usize,
     old_len: usize,
     pred: F,
+    panic_flag: bool,
 }
```

Comment:
This has grown enough fields that they should really be documented. Either individually here, or with a high-level description of what we're trying to do in the implementation of `next`.

Reply:
I added docs to each field in the latest commit.